asserts/asserts_test.go

```diff
@@ -164,6 +169,99 @@ func (as *assertsSuite) TestDecodeInvalid(c *C) {
 	}
 }
 
+func checkContent(c *C, a asserts.Assertion, encoded string) {
+	expected, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+	expectedCont, _ := expected.Signature()
+
+	cont, _ := a.Signature()
+	c.Check(cont, DeepEquals, expectedCont)
+}
+
+func (as *assertsSuite) TestDecoderHappy(c *C) {
+	stream := new(bytes.Buffer)
+	stream.WriteString(exampleEmptyBody2NlNl)
+	stream.WriteString("\n")
+	stream.WriteString(exampleBodyAndExtraHeaders)
+	stream.WriteString("\n")
+	stream.WriteString(exampleEmptyBodyAllDefaults)
+	stream.WriteString("\n\n")
```

Comment:
that's what an Encoder would need to do, which it seems I need to write actually, I need it in another place